tests/utils/streams/mixins.py

```diff
@@ -0,0 +1,159 @@
+import uuid
+from abc import ABC, abstractmethod
+from concurrent.futures import wait
+from contextlib import closing
+from typing import ContextManager, Mapping, Sequence
+
+import pytest
+
+from snuba.utils.streams.consumer import Consumer, ConsumerError, EndOfPartition
+from snuba.utils.streams.producer import Producer
+from snuba.utils.streams.types import Message, Partition, Topic
+
+
+class StreamsTestMixin(ABC):
```

Comment:
I will use this with the `DummyConsumer` and merge it's tests in soon: https://github.com/getsentry/snuba/blob/4dedea4d298d5c5fece7cd7cbb6528ebb2c2c2f0/tests/utils/streams/test_dummy.py#L8-L52